### PR TITLE
GH-4896 LMDB: Update benchmarks and improve performance

### PR DESCRIPTION
**Pull request by kenwenzel**
GitHub issue resolved: #4896 

Briefly describe the changes proposed in this PR:

Use same benchmarks as NativeStore and improve performance of value comparisons.

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by kenwenzel** [comment]
QueryBenchmark:
```
LmdbStore:
Benchmark                                                     Mode  Cnt     Score     Error  Units
QueryBenchmark.complexQuery                                   avgt    5    32.971 ±   6.779  ms/op
QueryBenchmark.different_datasets_with_similar_distributions  avgt    5    16.640 ±   1.517  ms/op
QueryBenchmark.groupByQuery                                   avgt    5     7.608 ±   1.243  ms/op
QueryBenchmark.long_chain                                     avgt    5  5738.926 ± 120.673  ms/op
QueryBenchmark.lots_of_optional                               avgt    5  1592.388 ±  66.201  ms/op
QueryBenchmark.minus                                          avgt    5  4842.102 ± 900.609  ms/op
QueryBenchmark.nested_optionals                               avgt    5  1265.482 ±  33.340  ms/op
QueryBenchmark.pathExpressionQuery1                           avgt    5   152.972 ±   8.200  ms/op
QueryBenchmark.pathExpressionQuery2                           avgt    5    24.418 ±   2.253  ms/op
QueryBenchmark.query_distinct_predicates                      avgt    5   443.126 ±  23.028  ms/op
QueryBenchmark.simple_filter_not                              avgt    5    48.037 ±   3.259  ms/op

NativeStore:
Benchmark                                                     Mode  Cnt     Score     Error  Units
QueryBenchmark.complexQuery                                   avgt    5    36.808 ±   3.314  ms/op
QueryBenchmark.different_datasets_with_similar_distributions  avgt    5    14.337 ±   1.436  ms/op
QueryBenchmark.groupByQuery                                   avgt    5    29.357 ±   6.870  ms/op
QueryBenchmark.long_chain                                     avgt    5  7252.237 ± 901.852  ms/op
QueryBenchmark.lots_of_optional                               avgt    5  1673.769 ± 189.591  ms/op
QueryBenchmark.minus                                          avgt    5  4349.451 ± 314.739  ms/op
QueryBenchmark.nested_optionals                               avgt    5  1655.361 ± 196.522  ms/op
QueryBenchmark.pathExpressionQuery1                           avgt    5   179.215 ±  14.714  ms/op
QueryBenchmark.pathExpressionQuery2                           avgt    5    29.894 ±   1.878  ms/op
QueryBenchmark.query_distinct_predicates                      avgt    5  1592.579 ± 127.726  ms/op
QueryBenchmark.simple_filter_not                              avgt    5    70.175 ±   2.252  ms/op
```

QueryWriteBenchmark:
```
LmdbStore:
Benchmark                                                    Mode  Cnt     Score      Error  Units
QueryWriteBenchmark.clear                                    avgt    3  2483.137 ±  962.811  ms/op
QueryWriteBenchmark.clearByQuery                             avgt    3  2447.176 ± 1341.499  ms/op
QueryWriteBenchmark.clearByQueryReadCommitted                avgt    3  2468.667 ± 1430.495  ms/op
QueryWriteBenchmark.clearContext                             avgt    3  2421.495 ±  634.775  ms/op
QueryWriteBenchmark.clearContextReadCommitted                avgt    3  2400.436 ±  615.277  ms/op
QueryWriteBenchmark.clearReadCommitted                       avgt    3  2483.619 ±  581.417  ms/op
QueryWriteBenchmark.removeByQuery                            avgt    3   366.599 ±  174.149  ms/op
QueryWriteBenchmark.removeByQueryReadCommitted               avgt    3   863.178 ±  498.291  ms/op
QueryWriteBenchmark.simpleUpdateQueryIsolationNone           avgt    3   604.452 ±  107.251  ms/op
QueryWriteBenchmark.simpleUpdateQueryIsolationReadCommitted  avgt    3   975.728 ±  514.399  ms/op

NativeStore:
Benchmark                                                    Mode  Cnt     Score       Error  Units
QueryWriteBenchmark.clear                                    avgt    3  3711.374 ±   338.500  ms/op
QueryWriteBenchmark.clearByQuery                             avgt    3  4988.268 ± 15639.497  ms/op
QueryWriteBenchmark.clearByQueryReadCommitted                avgt    3  4390.204 ±  1864.102  ms/op
QueryWriteBenchmark.clearContext                             avgt    3  3684.772 ±  2865.683  ms/op
QueryWriteBenchmark.clearContextReadCommitted                avgt    3  4129.391 ±  7993.073  ms/op
QueryWriteBenchmark.clearReadCommitted                       avgt    3  4586.808 ±   473.664  ms/op
QueryWriteBenchmark.removeByQuery                            avgt    3   731.911 ±   515.657  ms/op
QueryWriteBenchmark.removeByQueryReadCommitted               avgt    3  1125.343 ±  3172.579  ms/op
QueryWriteBenchmark.simpleUpdateQueryIsolationNone           avgt    3  1113.340 ±  1283.696  ms/op
QueryWriteBenchmark.simpleUpdateQueryIsolationReadCommitted  avgt    3  1070.583 ±   778.378  ms/op
```

File: core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/benchmark/QueryBenchmark.java [hunk 1]
```diff
@@ -151,100 +160,140 @@ public long complexQuery() {
...
-	public long distinctPredicatesQuery() {
+	public long pathExpressionQuery1() {
 
 		try (SailRepositoryConnection connection = repository.getConnection()) {
 			return connection
-					.prepareTupleQuery(query5)
+					.prepareTupleQuery(query7_pathexpression1)
 					.evaluate()
 					.stream()
 					.count();
```
Comment (hmottestad) [inline core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/benchmark/QueryBenchmark.java:170]: I believe I changed this code a bit in the memory store. The .count() method doesn't call close. I made a small helper method to simplify the code a bit:

```
	private static long count(TupleQueryResult evaluate) {
		try (Stream<BindingSet> stream = evaluate.stream()) {
			return stream.count();
		}
	}
```

Then you can use it like this:

```
@Benchmark
	public long complexQuery() {
		try (SailRepositoryConnection connection = repository.getConnection()) {
			return count(connection
					.prepareTupleQuery(query4)
					.evaluate()
			);
		}
	}
```

**Review by hmottestad — CHANGES_REQUESTED** [review]
Looks good. If want to fix the closing of the streams that would be nice. You should have the permission to go ahead and merge once your are happy with everything :)

File: core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/benchmark/QueryBenchmark.java (hunk 1 above)
Comment (kenwenzel) [reply core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/benchmark/QueryBenchmark.java]: I've fixed this.

**Comment by kenwenzel** [comment]
Fixed by https://github.com/eclipse-rdf4j/rdf4j/pull/4897

**Pull request merged**